modules/oe_translation_cdt/src/ContentFormatter/XmlFormatter.php

```diff
@@ -0,0 +1,184 @@
+
+    $entity_url = $request->getContentEntity()?->toUrl('canonical', ['absolute' => TRUE])->toString();
+    $transaction_item = (new TransactionItem())
+      ->setPreviewLink('NO PREVIEW LINK')
+      ->setSourceReference((string) $entity_url)
+      ->setTransactionItemFields($transaction_item_fields);
+
+    $transaction = (new Transaction())
+      ->setTransactionId((string) $request->id())
+      ->setTransactionCode('Drupal Translation Request')
+      ->setDrupalVersion(\Drupal::VERSION)
+      ->setModuleVersion($this->moduleExtensionList->getExtensionInfo('oe_translation')['version'])
+      ->setProducerDateTime(\DateTime::createFromFormat('U', (string) $this->time->getCurrentTime()))
+      ->setTotalCharacterLength($character_count)
+      ->setTransactionItems([$transaction_item]);
+
+    return $this->serializer()->serialize(
+      data: [
+        '@xmlns:xsi' => 'http://www.w3.org/2001/XMLSchema-instance',
+        '@xsi:noNamespaceSchemaLocation' => '//static.cdt.europa.eu/webtranslations/schemas/Drupal-Translation-V8-2.xsd',
+        '#' => $transaction,
+      ],
+      format: XmlEncoder::FORMAT,
+      context: [
+        'xml_root_node_name' => 'Transaction',
+        'xml_version' => '1.0',
+        'xml_encoding' => 'UTF-8',
+        'xml_standalone' => 'yes',
+        'xml_format_output' => TRUE,
+      ]
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function import(string $file, TranslationRequestInterface $request): array {
+    // Flatted the entire original request data so we have the original bits
+    // as well to combine with the translation values.
+    $flattened_request_data = TranslationSourceHelper::flatten($request->getData());
+
+    // Start a fresh array of translation data so that we can only include
+    // things that actually have translation values. This is to avoid
+    // duplicating a lot of storage data.
+    $translation_data = [];
+
+    /** @var \Drupal\oe_translation_cdt\Model\Transaction $transaction */
+    $transaction = $this->serializer()->deserialize(
+      data: $file,
+      type: Transaction::class,
+      format: XmlEncoder::FORMAT,
+      context: [
+        'xml_root_node_name' => 'Transaction',
+      ]
+    );
+    $transaction_item = $transaction->getTransactionItems()[0] ?? NULL;
+    $fields = $transaction_item ? $transaction_item->getTransactionItemFields() : [];
+
+    foreach ($fields as $field) {
+      $key = trim($field->getResourceName(), '[]');
+      if (!isset($flattened_request_data[$key])) {
+        continue;
+      }
+      $translation_data[$key] = $flattened_request_data[$key];
+      $translation_data[$key]['#translation'] = ['#text' => $field->getContent()];
+    }
+
+    return TranslationSourceHelper::unflatten($translation_data);
+  }
+
+  /**
+   * Creates and returns the serializer on-demand.
+   *
+   * @return \Symfony\Component\Serializer\Serializer
+   *   The serializer.
+   */
+  protected function serializer(): Serializer {
```

Comment:
I have used this type of design pattern to avoid having to create the serializer if it is not needed at all. In this case, the `serialization' module is not required.

Previous modules used TWIG and SimpleXML for serialization, but the Symfony approach is much more testable and also allows data import.